powertools-kafka/src/test/java/software/amazon/lambda/powertools/kafka/serializers/KafkaProtobufDeserializerTest.java

```diff
@@ -107,11 +107,33 @@ void shouldDeserializeProtobufDataWithComplexMessageIndex() throws IOException {
                 .setPrice(299.99)
                 .build();
 
-        // Create protobuf data with complex message index (array [1,0])
+        // Create protobuf data with complex message index (array [2,2])
         byte[] protobufDataWithComplexIndex = createProtobufDataWithComplexMessageIndex(product);
 
         // When
-        TestProduct result = deserializer.deserializeObject(protobufDataWithComplexIndex, TestProduct.class);
+        TestProduct result = deserializer.deserializeObject(protobufDataWithComplexIndex, TestProduct.class, AbstractKafkaDeserializer.SchemaRegistryType.CONFLUENT);
+
+        // Then
+        assertThat(result).isNotNull();
+        assertThat(result.getId()).isEqualTo(789);
+        assertThat(result.getName()).isEqualTo("Complex Index Product");
+        assertThat(result.getPrice()).isEqualTo(299.99);
+    }
+
+    @Test
+    void shouldDeserializeProtobufDataWithSimpleMessageIndexConfluent() throws IOException {
+        // Given
+        TestProduct product = TestProduct.newBuilder()
+                .setId(789)
+                .setName("Complex Index Product")
+                .setPrice(299.99)
+                .build();
+
+        // Create protobuf data with complex message index (array [2,2])
```

Comment:
This comment needs to be updated.

Reply:
https://github.com/aws-powertools/powertools-lambda-java/pull/1907/commits/f4f66dd0507dcde26c59d89cbbd31f346d559ac7